apollo-router/tests/integration_tests.rs

```diff
@@ -450,20 +466,70 @@ async fn persisted_queries() {
     let (router, registry) = setup_router_and_registry_with_config(config).await.unwrap();
 
     // Successfully run a persisted query.
-    let actual = query_with_router(router.clone(), pq_request(PERSISTED_QUERY_ID)).await;
+    let actual = query_with_router(router.clone(), pq_request(PERSISTED_QUERY_ID, None)).await;
+    assert!(actual.errors.is_empty());
+    assert_eq!(actual.data.as_ref(), Some(&expected_data));
+    assert_eq!(
+        registry.totals_reset(),
+        hashmap! {"accounts".to_string() => 1}
+    );
+
+    // Successfully run a persisted query with client name that has its own operation.
+    let actual = query_with_router(
+        router.clone(),
+        pq_request(
+            PERSISTED_QUERY_ID,
+            Some(("apollographql-client-name", "mobile")),
+        ),
+    )
+    .await;
+    assert!(actual.errors.is_empty());
+    assert_eq!(actual.data.as_ref(), Some(&expected_data_mobile));
+    assert_eq!(
+        registry.totals_reset(),
+        hashmap! {"accounts".to_string() => 1}
+    );
+
+    // Successfully run a persisted query with client name that has its own operation,
```

Comment:
I'm not sure how to add an integration test that tests the "client name read from the header named by `telemetry.apollo.client_name_header` because I'm not sure if there's an easy way to enable the part of the telemetry code that reads the header without setting up a full reporting agent situation.  (Maybe it would be OK for the PQ code to directly read this config field from the configuration instead? I had trouble getting that to work though.)

Reply:
There are other integration tests that actually spin up a router, see the `apollo-router/tests` directory. This enables a full testing of a feature including telemetry.

Reply:
That is the directory I'm in?

That said, this becomes trivial if I'm able to make the change mentioned above where the telemetry config is respected even when telemetry isn't enabled.